packages/headless/src/features/commerce/recommendations/recommendations-actions.ts

```diff
@@ -19,11 +19,20 @@ export interface QueryRecommendationsCommerceAPIThunkReturn {
 
 const buildRecommendationCommerceAPIRequest = async (
   slotId: string,
-  state: StateNeededByQueryCommerceAPI
+  state: StateNeededByQueryCommerceAPI,
+  productId?: string
 ): Promise<CommerceRecommendationsRequest> => {
   const commerceAPIRequest = await buildBaseCommerceAPIRequest(state, slotId);
   return {
     ...commerceAPIRequest,
+    context: {
+      ...commerceAPIRequest.context,
+      product: productId
+        ? {
+            productId,
+          }
+        : undefined,
```

Comment:
Instead of setting the`product` to `undefined`, we should conditionally include `product`
```suggestion
      ...(productId ? { product: { productId }} : {}),
```